test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +29,125 @@ def test_password_hack(static_proxy):
+            assert link_auth_uri == local_link_proxy.link_auth_uri, \
+                f"Line='{line}' should contain a http auth link of form '{local_link_proxy.link_auth_uri}/<psql_session_id>'."
+            break
+        log.debug("line %d does not contain expected result: %s", line_id, line)
+
+    assert line_id <= max_num_lines_of_welcome_message, "exhausted given attempts, did not get the result"
+    assert psql_session_id is not None, "psql_session_id not found from output of proc.stderr.readline()"
+    log.info(f"Got psql_session_id={psql_session_id=}")
+
+    return psql_session_id
+
+
+def create_and_send_db_inf(local_vanilla_pg, psql_session_id, mgmt_port):
+    pg_user = "proxy"
+    pg_password = "password"
+
+    local_vanilla_pg.start()
+    query = "create user " + pg_user + " with login superuser password '" + pg_password + "'"
+    local_vanilla_pg.safe_psql(query)
+
+    port = local_vanilla_pg.default_options['port']
+    host = local_vanilla_pg.default_options['host']
+    dbname = local_vanilla_pg.default_options['dbname']
+
+    db_info_dict = {
+        "session_id": psql_session_id,
+        "result": {
+            "Success": {
+                "host": host,
+                "port": port,
+                "dbname": dbname,
+                "user": pg_user,
+                "password": pg_password
+            }
+        }
+    }
+    db_info_str = json.dumps(db_info_dict)
+    cmd_line_args__to__mgmt = [
+        "psql",
+        "-h",
+        "127.0.0.1",  # localhost
+        "-p",
+        f"{mgmt_port}",
+        '-c',
+        db_info_str
+    ]
+
+    log.info(f"Sending to proxy the user and db info: {cmd_line_args__to__mgmt}")
+    p = subprocess.Popen(cmd_line_args__to__mgmt, stdout=subprocess.PIPE)
+    out, err = p.communicate()
+    assert "ok" in str(out)
+
+
+@pytest.mark.asyncio
+async def test_psql_session_id(vanilla_pg, link_proxy):
+    """
+    Test copied and modified from: test_project_psql_link_auth test from cloud/tests_e2e/tests/test_project.py
+     Step 1. establish connection to the proxy
+     Step 2. retrieves session_id
+     Step 3. create a vanilla_pg and send user and db info via command line (using Popen) a psql query via mgmt port to proxy.
+     Step 4. assert that select 1 has been executed correctly.
+    """
+
+    # Step 1.
+    psql = PSQL(
+        host=link_proxy.host,
+        port=link_proxy.proxy_port,
+    )
+    proc = await psql.run("select 1")
+
+    # Step 2.
+    psql_session_id = await get_session_id_from_welcome_message(link_proxy, proc)
+
+    # Step 3.
+    create_and_send_db_inf(vanilla_pg, psql_session_id, link_proxy.mgmt_port)
+
+    # Step 4.
+    # Expecting proxy output::
+    # b' ?column? \n'
+    # b'----------\n'
+    # b'        1\n'
+    # b'(1 row)\n'
+    max_num_output_lines = 4
+    str_sum = ""
+    for i in range(max_num_output_lines):
+        raw_bytes_line = await proc.stdout.readline()
+        str_sum += str(raw_bytes_line)
```

Comment:
Is `proc.stdout` a bytes? If so, can we get the string content with `proc.stdout.decode('utf-8')`? Seems silly to split to lines only to merge them again.

Or we can avoid converting to string and directly compare it with a bytes literal `b'something something'`

P.S: I think `str(some_bytes)` will not decode the bytes to str, but instead give you a string representation for debugging.

Reply:
done